tests/lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,316 @@
+          line: 6
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup>
+          console.log('test1')
+          const props = defineProps({
+            test: Boolean
+          })
+          console.log('test2')
+          const emit = defineEmits(['update:test'])
+          console.log('test3')
+        </script>
+      `,
+      output: `
+        <script setup>
+          const emit = defineEmits(['update:test'])
+          const props = defineProps({
+            test: Boolean
+          })
+          console.log('test1')
+          console.log('test2')
+          console.log('test3')
+        </script>
+      `,
+      options: optionsEmitsFirst,
+      errors: [
+        {
+          message: message('defineEmits'),
+          line: 8
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script lang="ts" setup>
+          interface Props {
+            msg?: string
+            labels?: string[]
+          }
+
+          const props = withDefaults(defineProps<Props>(), {
+            msg: 'hello',
+            labels: () => ['one', 'two']
+          })
+          const emit = defineEmits<{(e: 'update:test'): void}>()
+        </script>
+      `,
+      output: `
+        <script lang="ts" setup>
+          interface Props {
+            msg?: string
+            labels?: string[]
+          }
+
+          const emit = defineEmits<{(e: 'update:test'): void}>()
+          const props = withDefaults(defineProps<Props>(), {
+            msg: 'hello',
+            labels: () => ['one', 'two']
+          })
+        </script>
+      `,
+      parserOptions: {
+        parser: require.resolve('@typescript-eslint/parser')
+      },
+      options: optionsEmitsFirst,
+      errors: [
+        {
+          message: message('defineEmits'),
+          line: 12
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script lang="ts" setup>
+          import bla from 'bla';
+          interface Foo {};
+          type Bar = {};
+          // <--- auto-fix should move \`defineProps\` here
+          const someOtherCode = '';
+          import foo from 'bar'; // not idiomatic, but allowed
+          interface SomeOtherInterface {};
+          defineProps({ test: Boolean });
+        </script>
+      `,
+      output: `
+        <script lang="ts" setup>
+          import bla from 'bla';
+          interface Foo {};
+          type Bar = {};
+          // <--- auto-fix should move \`defineProps\` here
+          defineProps({ test: Boolean });
+          const someOtherCode = '';
+          import foo from 'bar'; // not idiomatic, but allowed
+          interface SomeOtherInterface {};
+        </script>
+      `,
+      parserOptions: {
+        parser: require.resolve('@typescript-eslint/parser')
+      },
+      errors: [
+        {
+          message: message('defineProps'),
+          line: 10
+        }
+      ]
+    }
```

Comment:
Add test with comments. I think that the comment just before the define macro needs to be moved together with the define macro.

```ts
          console.log('test')
          /** Description for props */
          const props = withDefaults(defineProps<Props>(), {
            msg: 'hello'
          })
          console.log('test')
          /** Description for emit */
          const emit = defineEmits<{(e: 'test'): void}>()
```